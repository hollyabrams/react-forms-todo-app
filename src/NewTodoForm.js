import { useState } from 'react';
import { v4 as uuid } from 'uuid';
import './NewTodoForm.scss';


function NewTodoForm({ addTodo }) {
  const [task, setTask] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    if (task.trim()) {
      const newTodo = { id: uuid(), task: task, completed: false };
      addTodo(newTodo);
      setTask('');
    }
  };

  const handleChange = (e) => {
    setTask(e.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="task">Task:</label>
      <input
        type="text"
        id="task"
        name="task"
        value={task}
        onChange={handleChange}
      />
      <button type="submit">Add Todo</button>
    </form>
  );
}

export default NewTodoForm;

