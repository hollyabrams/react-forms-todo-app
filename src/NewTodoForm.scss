form {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 20px 0;

    label {
        font-size: 18px;
        margin-bottom: 10px;
    }

    input {
        width: 300px;
        padding: 10px;
        border: 1px solid gray;
        border-radius: 5px;
        margin-bottom: 20px;
        font-size: 18px;
    }

    button {
        padding: 10px 20px;
        background-color: #4caf50;
        color: white;
        border: none;
        border-radius: 5px;
        font-size: 18px;
        cursor: pointer;
        transition: all 0.3s ease;

        &:hover {
            background-color: #3e8e41;
        }
    }
}