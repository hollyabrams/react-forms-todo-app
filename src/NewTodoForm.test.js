import { render, screen } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

describe('NewTodoForm', () => {
  it('should render without errors', () => {
    render(<NewTodoForm addTodo={() => {}} />);
    expect(screen.getByLabelText('Task:')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Add Todo' })).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<NewTodoForm addTodo={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
