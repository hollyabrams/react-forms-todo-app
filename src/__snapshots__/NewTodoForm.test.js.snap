// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm should match snapshot 1`] = `
<div>
  <form>
    <label
      for="task"
    >
      Task:
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</div>
`;
